index.js

```diff
@@ -242,16 +243,20 @@ NYC.prototype.instrumentAllFiles = function (input, output, cb) {
 }
 
 NYC.prototype.walkAllFiles = function (dir, visitor) {
-  var pattern = null
-  if (this.extensions.length === 1) {
-    pattern = '**/*' + this.extensions[0]
-  } else {
-    pattern = '**/*{' + this.extensions.join() + '}'
-  }
+  const pattern = (this.extensions.length === 1)
+    ? `**/*${this.extensions[0]}`
+    : `**/*{${this.extensions.join()}}`
 
-  glob.sync(pattern, { cwd: dir, nodir: true, ignore: this.exclude.exclude }).forEach(function (filename) {
-    visitor(filename)
+  let filesToWalk = glob.sync(pattern, { cwd: dir, nodir: true, ignore: this.exclude.exclude })
+
+  // package node-glob no longer observes negated excludes, so we need to restore these files ourselves
+  const excludeNegatedPaths = this.exclude.excludeNegated
+  excludeNegatedPaths.forEach(pattern => {
+    filesToWalk = filesToWalk.concat(glob.sync(pattern, { cwd: dir, nodir: true }))
   })
+  filesToWalk = arrayUniq(filesToWalk)
```

Comment:
To avoid the one time use dependency and the multiple operations on `filesToWalk`, I'd write it something like this. Though yours is a bit clearer in terms of intentions so 🤷‍♂️ 

```js
const excludeNegatedPaths = this.exclude.excludeNegated.map(pattern => glob.sync(pattern, { cwd: dir, nodir: true })

// do the walkAllFiles thing
return glob.sync(pattern, { cwd: dir, nodir: true, ignore: this.exclude.exclude })
  .concat(excludeNegatedPaths)
  .filter((value, index, self) => self.indexOf(value) === index)
  .forEach(visitor)
```

Reply:
Yeah, I felt a bit bad adding arrayUniq as a dependency as it's so simple too that it would be easier to define it inline than add it to package.json.  I figured promoting it may not be such an issue since it was already a sub dependency, that said I'm certainly not against defining it inline as
```javascript
const arrayUniq = x => [...new Set(x)];
```
I did have a go at removing the `let` in there but I got to a point where I felt I was being a bit pedantic and should probably just leave it be.  Anyway I'll have another look and see if I can make it a bit nicer, and remove the dep, drop the let and return the results.

Reply:
I don't think we have access to the spread operator for node 6

Reply:
Actually we do have access to array spread and rest arguments.  So `[...new Set(x)]` works in node 6, as does `function (...args) {}`.  Object spread is not available until node.js 8 (so we cannot do things like `{...defaultOpts, ...options}`, for that we still have to use `Object.assign`.

Reply:
Deferred for now so we can look at merging what we have.